android/BOINC/app/src/main/java/edu/berkeley/boinc/client/Monitor.kt

```diff
@@ -331,17 +331,17 @@ class Monitor : LifecycleService() {
                 val transfers = clientInterface.fileTransfers
                 val acctMgrInfo = clientInterface.acctMgrInfo
                 val newNotices = clientInterface.getNotices(clientStatus.mostRecentNoticeSeqNo)
+                var nullValues = ""
+                if (state == null) {
+                    nullValues += "state "
+                    Log.e(Logging.TAG, "readClientStatus(): connection problem, null: " + nullValues);
+                }
                 if (allNotNull(status, state, state!!.hostInfo, acctMgrInfo)) {
                     clientStatus.setClientStatus(status, state.results, state.projects,
                             transfers, state.hostInfo, acctMgrInfo,
                             newNotices)
                 } else {
-                    var nullValues = ""
-                    if (state == null) {
-                        nullValues += "state "
-                    } else {
-                        if (state.hostInfo == null) nullValues += "state.host_info "
-                    }
+                    if (state.hostInfo == null) nullValues += "state.host_info "
```

Comment:
This will fail if `state == null`

Reply:
No. It will fall before. (That why it have the warning). At this point of the code the state will never be null.
if state will be null. it will never enter inside the if. look here:
https://github.com/BOINC/boinc/pull/3946/files#diff-e5abe666500a14899c926f7fb6a9a9bbL334
`if (allNotNull(status, state, state!!.hostInfo, acctMgrInfo)) {`  we put inside `state!!.hostInfo`.
We can change the function that will not get the state.info, and take it from inside. (and then check if state is null inside).